assistant/v2.ts

```diff
@@ -76,16 +76,24 @@ class AssistantV2 extends BaseService {
    * @param {Function} [callback] - The callback that handles the response.
    * @returns {NodeJS.ReadableStream|void}
    */
-  public createSession(params: AssistantV2.CreateSessionParams, callback?: AssistantV2.Callback<AssistantV2.SessionResponse>): NodeJS.ReadableStream | void {
+  public createSession(params: AssistantV2.CreateSessionParams, callback?: AssistantV2.Callback<AssistantV2.SessionResponse>): NodeJS.ReadableStream | Promise<any> | void {
     const _params = extend({}, params);
-    const _callback = (callback) ? callback : () => { /* noop */ };
+    const _callback = (callback) ? callback : undefined;
     const requiredParams = ['assistant_id'];
 
     const missingParams = getMissingParams(_params, requiredParams);
     if (missingParams) {
       return _callback(missingParams);
```

Comment:
Missing parameters won't get caught in a `reject()`.

Reply:
Check that if we have errors in the previous statements we always return a reject.

Reply:
Great catch, I feel silly for missing this. All I have to do is move the Promise return function to before the missing parameter check and it will call the reject properly